bcipy/display/paradigm/vep/layout.py

```diff
@@ -0,0 +1,174 @@
+            square (ex. 4, 9, 16, 25)
+        colors - tuple of color names between which to alternate
+        center - position of the center point of the board
+        board_size - size in layout units of the entire checkerboard; square
+            size will be computed from this.
+    """
+    boxes_per_row = int(sqrt(squares))
+    assert boxes_per_row**2 == squares, "Must be a perfect square"
+
+    square_width = board_size[0] / boxes_per_row
+    square_height = board_size[1] / boxes_per_row
+    square_size = (square_width, square_height)
+    center_x, center_y = center
+
+    # find the center position of the left_top square
+    move_count = int(boxes_per_row / 2)
+    if squares % 2 == 0:
+        # with an even number the center is on a vertex; adjust by half a box
+        move_count -= 0.5
+    left = left_of(center_x, square_width * move_count)
+    top = above(center_y, square_height * move_count)
+
+    # iterate starting at the top left and proceeding in a zig zag
+    # pattern to correspond with alternating checkerboard colors.
+    positions = []
+    for row in range(boxes_per_row):
+        if row > 0:
+            top = below(top, square_height)
+        for col in range(boxes_per_row):
+            positions.append((left, top))
+            if col < boxes_per_row - 1:
+                if row % 2 == 0:
+                    left = right_of(left, square_width)
+                else:
+                    left = left_of(left, square_width)
+    return [
+        CheckerboardSquare(*args)
+        for args in zip(positions, cycle(colors), cycle([square_size]))
+    ]
+
+
+class BoxConfiguration():
+    """Computes box size and positions for a VEP display.
+
+    In this configuration, there is one row on top and one on the bottom.
+
+    Parameters
+    ----------
+        num_boxes - number of boxes; currently supports 4 or 6.
+        layout - defines the boundaries within a Window in which the text areas
+            will be placed
+        spacing_pct - used to specify spacing between boxes within a row.
+        row_pct - what percentage of the height each row should use.
+    """
+
+    def __init__(self,
+                 layout: Layout,
+                 num_boxes: int,
+                 spacing_pct: Optional[float] = None,
+                 height_pct: float = 0.25):
+        assert num_boxes == 4 or num_boxes == 6, 'Number of boxes must be 4 or 6'
+        assert height_pct <= 0.5, "Rows can't take more than 50% of the height"
+        self.num_boxes = num_boxes
+        self.layout = layout
+
+        default_spacing = {4: 0.1, 6: 0.05}
+        if not spacing_pct:
+            spacing_pct = default_spacing[num_boxes]
+        self.spacing_pct = spacing_pct
+        self.height_pct = height_pct
+        self._row_count = 2
+
+    def validate(self):
+        """Validate invariants"""
+        assert self.num_boxes == 4 or self.num_boxes == 6, 'Number of boxes must be 4 or 6'
```

Comment:
Is this still needed?

Reply:
It's needed because it's possible to change these values between constructing the config and getting the positions. However, the code was repeated. I refactored to have the constructor use this method as well.